services/QuillLMS/app/services/classlink_integration/sso_user_importer.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module ClasslinkIntegration
+  class SsoUserImporter < ApplicationService
+    attr_reader :tenant_id, :user_sourced_id, :access_token, :classlink_instance
+
+    class RunError < StandardError; end
+
+    def initialize(tenant_id:, user_sourced_id:, access_token:, classlink_instance:)
+      @tenant_id = tenant_id
+      @user_sourced_id = user_sourced_id
+      @access_token = access_token
+      @classlink_instance = classlink_instance
+    end
+
+    def run
+      account = ClasslinkAccount.find_by(classlink_instance:, external_id: user_sourced_id)
+
+      return account if account
+
+      instance = DistrictsImporter.find_or_create_district_and_instance(
+        {
+          'name' => "#{tenant_id} District",
+          'tenant_id' => tenant_id,
+          'tenant_name' => "#{tenant_id} District",
+
+        }
+      )
+
+      incoming_user = UsersFetcher.new(
+        district_oneroster_id: nil,
+        district_bearer_token: nil,
+        district_id: nil
+      ).transform_api_response(fetch)
```

Comment:
You could consider making this a service:

```ruby

UserFetcher.run(
  district_oneroster_id: nil,
  district_bearer_token: nil,
  district_id: nil,
  api_response:
)

Reply:
`UsersFetcher` is already an ApplicationService. I am side calling one of it's methods here, in order to maximize shared code.